addons/mrp_subcontracting/models/stock_warehouse.py

```diff
@@ -41,6 +54,14 @@ def get_rules_dict(self):
             })
         return result
 
+    def _update_global_route_resupply_subcontractor(self):
+        route_id = self._find_global_route('mrp_subcontracting.route_resupply_subcontractor_mto',
+                                           _('Resupply Subcontractor on Order'))
+        if not route_id.sudo().rule_ids:
```

Comment:
Adding `sudo()` includes getting the archived rules. You have to do a sudo search for the rules to avoid the archive rules while getting them for all companies.